Introducing QuartusFlow (#56)
SpinalHDL/SpinalDoc-RTD: source/SpinalHDL/Libraries/EDA/altera/quartus_flow.rst

QuartusFlow
===========

Introduction
------------

A compilation flow is an Altera-defined sequence of commands that use a combination of command-line executables.
A full compilation flow launches all Compiler modules in sequence to synthesize, fit, analyze final timing, and generate a device programming file. 

> Tools in `this file <https://github.com/SpinalHDL/SpinalHDL/blob/dev/lib/src/main/scala/spinal/lib/eda/altera/QuartusFlow.scala>`_ help you get rid of redundant Quartus GUI.

Reviewer: This should be an anonymous link.
```suggestion
Tools in `this file <https://github.com/SpinalHDL/SpinalHDL/blob/dev/lib/src/main/scala/spinal/lib/eda/altera/QuartusFlow.scala>`__ help you get rid of redundant Quartus GUI.
```

For a single rtl file
---------------------

> The object ``spinal.lib.eda.altera.QuartusFlow`` can automatically report the area and maximum frequence of a single rtl file.

Reviewer: ```suggestion
The object ``spinal.lib.eda.altera.QuartusFlow`` can automatically report the used area and maximum frequency of a single rtl file.
```

Example
^^^^^^^

.. code-block:: scala

   val report = QuartusFlow(
      quartusPath="/eda/intelFPGA_lite/17.0/quartus/bin/",
      workspacePath="/home/spinalvm/tmp",
      toplevelPath="TopLevel.vhd",
      family="Cyclone V",
      device="5CSEMA5F31C6",
      frequencyTarget = 1 MHz
   )
   println(report)

> The code above will create a new Quartus project with ``TopLevel.vhd``. By the way, if you use the device ``EP4CE6E22C8N``, the family should be ``Cyclone IV E``.

Reviewer: It's fine as is (IMHO).. but I think in the feature is better to have a table with family and most used device serial.

Author: maybe there can link to altera website?

Author: I haven't found any 😞

Author: The `QuartusFlow` don't use the `family` or `device`. These two strings are directly passed to Quartus command line.

Reviewer: Maybe we need just a note like this:
```suggestion
The code above will create a new Quartus project with ``TopLevel.vhd``.

.. note::
   The ``family`` and ``device`` values are passed straight to the Quartus CLI as parameters. Please check the Quartus documentation for the correct value to use in your project.
```

what do you think?

Author: Fantastic！I'm learning a lot English from you. 😂

.. warning::
   This operation will remove the folder ``workspacePath``!

For an existing project
-----------------------

> The class ``spinal.lib.eda.altera.QuartusProject`` can automatically find configuration files in an existing project that are used to call a full compilation and program the device. 

Reviewer: I split the phrase. What do you think?
```suggestion
The class ``spinal.lib.eda.altera.QuartusProject`` can automatically find configuration files in an existing project. Those are used for compilation and programming the device. 
```

Author: yes! I speak broken English...
